src/models/event-timeline-set.js

```diff
@@ -149,13 +149,14 @@ EventTimelineSet.prototype.replaceEventId = function(oldEventId, newEventId) {
  * <p>This is used when /sync returns a 'limited' timeline.
  *
  * @param {string=} backPaginationToken   token for back-paginating the new timeline
- * @param {?bool} flush  Whether to flush the non-live timelines too.
```

Comment:
Right ignore me, we call `this.opts.canResetEntireTimeline` still, and just ternary it to set the token or not.